doc/source/cluster/kubernetes/examples/gpu-training-example.md

```diff
@@ -0,0 +1,150 @@
+
+If you are new to Kubernetes and you are planning to deploy Ray workloads on a managed
+Kubernetes service, we recommend taking a look at this {ref}`introductory guide <kuberay-k8s-setup>` first.
+
+It is not necessary to run this example with a cluster having that much RAM (>30GB per node in the following commands). Feel free to update
+the option `machine-type` and the resource requirements in `ray-cluster.gpu.yaml`.
+
+In the first command, we create a Kubernetes cluster `gpu-cluster-1` with one CPU node (`e2-standard-8`: 8 vCPU; 32 GB RAM). In the second command,
+we add a new node (`n1-standard-8`: 8 vCPU; 30 GB RAM) with a GPU (`nvidia-tesla-t4`) to the cluster.
+
+```shell
+# Step 1: Set up a Kubernetes cluster on GCP.
+# e2-standard-8 => 8 vCPU; 32 GB RAM
+gcloud container clusters create gpu-cluster-1 \
+    --num-nodes=1 --min-nodes 0 --max-nodes 1 --enable-autoscaling \
+    --zone=us-central1-c --machine-type e2-standard-8
+
+# Create a node-pool for GPU
+# n1-standard-8 => 8 vCPU; 30 GB RAM
+gcloud container node-pools create gpu-node-pool \
+  --accelerator type=nvidia-tesla-t4,count=1 \
+  --zone us-central1-c --cluster gpu-cluster-1 \
+  --num-nodes 1 --min-nodes 0 --max-nodes 1 --enable-autoscaling \
+  --machine-type n1-standard-8
+
+# Install NVIDIA GPU device driver
+kubectl apply -f https://raw.githubusercontent.com/GoogleCloudPlatform/container-engine-accelerators/master/nvidia-driver-installer/cos/daemonset-preloaded.yaml
+```
+
+## Step 2: Deploy a Ray cluster on Kubernetes with the KubeRay operator.
+
+To execute the following steps, please make sure you are connected to your Kubernetes cluster. (For GCP, you can do so by running the relevant `gcloud containers` command.)
+The first command will deploy KubeRay (ray-operator) to your Kubernetes cluster. The second command will create a ray cluster with the help of KubeRay.
+
+The third command is used to map port 8265 of the `ray-head` pod to **127.0.0.1:8265**. You can check
+**127.0.0.1:8265** to see the dashboard. The last command is used to test your Ray cluster by submitting a simple job.
+It is optional.
+
+```shell
+# Step 2: Deploy a Ray cluster on Kubernetes with the KubeRay operator.
+# Create the KubeRay operator
+kubectl create -k "github.com/ray-project/kuberay/ray-operator/config/default?ref=v0.3.0&timeout=90s"
+
+# Create a Ray cluster
+kubectl apply -f https://raw.githubusercontent.com/ray-project/ray/master/doc/source/cluster/kubernetes/configs/ray-cluster.gpu.yaml
+
+# port forwarding
+kubectl port-forward services/raycluster-head-svc 8265:8265
+
+# Test cluster (optional)
+ray job submit --address http://localhost:8265 -- python -c "import ray; ray.init(); print(ray.cluster_resources())"
+```
+
+## Step 3: Run the PyTorch image training benchmark.
```

Comment:
At the beginning of this, there should be a little explanation what this is doing. Consider embedding `pytorch_training_e2e_submit.py` into the documentation :)

Reply:
Good idea -- we can do something similar to what's done here: https://docs.ray.io/en/latest/cluster/kubernetes/examples/ml-example.html#submit-the-workload
https://github.com/ray-project/ray/blame/45788244f9b48f619840fb8f57f429416e68f133/doc/source/cluster/kubernetes/examples/ml-example.md#L95

Reply:
Updated https://github.com/ray-project/ray/pull/28353/commits/0f3aad56533171c9b6246b9cec5931191af42253.